numpy/meson.build

```diff
@@ -143,12 +145,62 @@ foreach subdir: pure_subdirs
   install_subdir(subdir, install_dir: np_dir)
 endforeach
 
-custom_target('__config__.py',
-  output: '__config__.py',
+compilers = {
+  'C': cc,
+  'CPP': cpp,
+  'CYTHON': meson.get_compiler('cython')
+}
+
+machines = {
+  'HOST': host_machine,
+  'BUILD': build_machine,
+}
+
+if meson.is_cross_build()
```

Comment:
Not sure how useful this is, please let me know if we need it.

Reply:
That does seem useful to me.

Reply:
Actually, I have to retract that. For cross-compilation, `build` and `host` should cover things. We don't really have a need for `target`. It's current also generating this warning:
```
numpy/meson.build:199: WARNING: The variable(s) 'TARGET_CPU', 'TARGET_CPU_ENDIAN', 'TARGET_CPU_FAMILY', 'TARGET_CPU_SYSTEM' in the input file 'numpy/__config__.py.in' are not present in the given configuration data.
```

The current output is of `show_config` is:
```
Machine Information:
  build:
    cpu: x86_64
    endian: little
    family: x86_64
    system: linux
  host:
    cpu: x86_64
    endian: little
    family: x86_64
    system: linux
```
That looks good. I'd suggest removing `target`, and adding one line:
```
cross-compiled: yes/no
```

Reply:
There's one more warning, that should be easy to fix:
```
numpy/meson.build:182: WARNING: Project targets '>= 0.64.0' but uses feature deprecated since '0.55.0': ExternalProgram.path. use ExternalProgram.full_path() instead
Configuring __config__.py using configuration
```

Reply:
Fixed in a35447fac005c8794ef620949a0448a8f4715251